MovieApp/PresentationLayer/MovieDetails/MovieDetailsViewController.swift

```diff
@@ -26,129 +27,113 @@ class MovieDetailsViewController: UIViewController {
-        presenter.getMovieCredits(
-            for: movieId,
-            maximumCrewMembers: noOfCrewRows * noOfCrewColumns) { [weak self] result in
-            guard let self = self else { return }
-
-            switch result {
-            case .success(let credits):
-                self.setCrewGridData(for: credits.crew)
-                self.hideEmptyCrewLabels()
-                self.castMembers = credits.cast
-                self.topBilledCastCollection.reloadData()
-            case .failure:
-                print("Failed to get movie credits")
-            }
-        }
-
-        presenter.getReview(for: movieId) { [weak self] result in
-            guard let self = self else { return }
-
-            switch result {
-            case .success(let reviewViewModels):
-                if reviewViewModels.count == 0 {
-                    self.showReviewsError(message: "No reviews available for this title.")
-                } else {
-                    self.setReviewData(for: reviewViewModels)
-                }
-            case .failure:
-                print("Failed to get movie reviews")
-            }
-        }
-
-        presenter.getRecommendations(for: movieId) { [weak self] result in
-            guard let self = self else { return }
-
-            switch result {
-            case .success(let recommendationViewModels):
-                self.recommendations = recommendationViewModels
-                self.recommendationCollection.reloadData()
-            case .failure:
-                print("Failed to get movie recommendations")
-            }
-        }
-    }
-
+    
     private func bindViews() {
-        navigationView.onBackButtonTap = { [weak self] in
-            guard let self = self else { return }
-
-            self.router?.goBack()
-        }
-
-        headerView.onFavoriteToggle = { movieId in
-            self.presenter.toggleFavorited(for: movieId) { [weak self] in
+        navigationView
+            .backButtonTapped
+            .sink { [weak self] _ in
                 guard let self = self else { return }
 
-                self.loadData(animated: false)
+                self.router?.goBack()
             }
-        }
-    }
-
-    private func showReviewsError(message: String) {
-        reviewsViewController.view.isHidden = true
-        noReviewsLabel.text = message
-        noReviewsLabel.isHidden = false
+            .store(in: &disposables)
+
+        headerView
+            .favoritedToggle
+            .sink(
+                receiveCompletion: { _ in },
+                receiveValue: { [weak self] movieId in
+                    self?.presenter.toggleFavorited(for: movieId)
+                })
+            .store(in: &disposables)
+        
+        presenter
+            .details(for: movieId)
+            .sink(
+                receiveCompletion: { _ in },
+                receiveValue: { [weak self] detailedMovieViewModel in
+                    guard let self = self else { return }
```

Comment:
you do not need to guard self

Reply:
I figured it would be better to guard `self` because inside these functions (`setMovieDetailsData`, `setCreditsData`, `setReviewData`, `setRecommendationData`) `self` is accessed multiple times.

Reply:
That does not matter. Once the function is called, self exists and cannot be deallocated.